packages/vscode-extension/src/dependency/DependencyManager.ts

```diff
@@ -69,24 +70,23 @@ export class DependencyManager implements Disposable, DependencyManagerInterface
     this.checkNodeModulesInstallationStatus();
 
     this.emitEvent("reactNative", {
-      status: npmPackageVersionCheck("react-native", MinSupportedVersion.reactNative),
+      status: npmPackageVersionCheck("react-native", appRoot, MinSupportedVersion.reactNative),
       isOptional: false,
     });
 
     this.emitEvent("expo", {
-      status: npmPackageVersionCheck("expo", MinSupportedVersion.expo),
-      isOptional: !shouldUseExpoCLI(),
+      status: npmPackageVersionCheck("expo", appRoot, MinSupportedVersion.expo),
+      isOptional: !shouldUseExpoCLI(appRoot),
     });
 
     this.checkProjectUsesExpoRouter();
     this.checkProjectUsesStorybook();
     this.checkEasCliInstallationStatus();
   }
 
-  public async validateNodeVersion() {
-    const appRootFolder = getAppRootFolder();
+  public async validateNodeVersion(appRoot: string) {
```

Comment:
given DependencyManager already stores appRoot, we shouldn't require it as argument.